tuf_conformance/test_file_download.py

```diff
@@ -238,3 +238,28 @@ def test_download_with_unknown_hash_algorithm(
     # and only then realize hash cannot be verified
     assert client.download_target(init_data, target_path) == 1
     assert client.get_downloaded_target_bytes() == []
+
+
+def test_artifact_cache(client: ClientRunner, server: SimulatorServer) -> None:
```

Comment:
Should we include the test where the artifact should be downloaded again because the contents changed?

Reply:
`test_multiple_changes_to_target` should handle that along with other cases... 

I can see some advantage from having a simple test that would only test what you suggest (instead of a long test that tests multiple things), but it does not seem strictly necessary 🤷

Reply:
Oh no I'm just not aware of what was in there and it was too much work from my phone to go looking. Lgtm.